src/cmd/logic-function.js

```diff
@@ -285,6 +359,103 @@ module.exports = class LogicFunctionsCommand extends CLICommandBase {
 		// TODO
 		console.log(org, name, id, saveTo);
 	}
+
+	async _setOrg(org) {
+		if (this.org === null) {
+			this.org = org;
+		}
+	}
+
+	async _getLogicFunctionList() {
+		if (this.logicFuncList === null) {
+			try {
+				const res = await this.api.getLogicFunctionList({ org: this.org });
+				this.logicFuncList = res.logic_functions;
+			} catch (e) {
+				throw createAPIErrorResult({ error: e, message: 'Error listing logic functions' });
+			}
+		}
+	}
+
+	async _getLogicFunctionData(id) {
+		try {
+			const logicFunction = await this.api.getLogicFunction({ org: this.org, id });
+			return logicFunction;
+		} catch (e) {
+			throw createAPIErrorResult({ error: e, message: 'Error getting logic function' });
+		}
+	}
+
+	_serializeLogicFunction(data) {
+		const logicFunctionCode = data.logic_function.source.code;
+		const logicFunctionConfigData = data.logic_function;
+		delete logicFunctionConfigData.source;
+
+		return { logicFunctionConfigData, logicFunctionCode };
+	}
+
+	async _generateFiles({ logicFunctionConfigData, logicFunctionCode, name }) {
+		// TODO: Address spaces in the name for Windows
+		const slugName = slugify(name);
+		const dirPath = path.join(process.cwd(), `${slugName}`);
+		const jsonPath = path.join(dirPath, `${slugName}.logic.json`);
+		const jsPath = path.join(dirPath, `${slugName}.js`);
+
+		await this._validatePaths({ dirPath, jsonPath, jsPath });
+
+		await fs.ensureDir(dirPath);
+		await fs.writeFile(jsonPath, JSON.stringify(logicFunctionConfigData, null, 2));
+		await fs.writeFile(jsPath, logicFunctionCode);
+
+		return { dirPath, jsonPath, jsPath };
+	}
+
+	async _getLogicFunctionIdAndName(name, id) {
+		if (!id && !name) {
+			name = await this._selectLogicFunction(this.logicFuncList);
+			id = this._getIdFromName(name, this.logicFuncList);
+		} else if (!id && name) {
+			id = this._getIdFromName(name, this.logicFuncList);
+		} else if (id && !name) {
+			name = this._getNameFromId(id, this.logicFuncList);
+		}
+
+		return { name, id };
+	}
+
+	async _selectLogicFunction(list) {
+		if (list.length === 0) {
+			throw new Error('No logic functions found');
+		}
+		const answer = await this._prompt({
+			type: 'list',
+			name: 'logic_function',
+			message: 'Which logic function would you like to download?',
+			choices : list,
+			nonInteractiveError: 'Provide name for the logic function'
+		});
+		return answer.logic_function;
```

Comment:
@monkbroc I know we discussed to return name and id here. I didn't make that change to keep the `_getLogicFunctionIdAndName` look consistent in the way it calls the `_getXFromY` methods - Kindly let me know if this should have been different.